src/app/task/task-details/task-details.component.ts

```diff
@@ -41,6 +55,47 @@ export class TaskDetailsComponent implements OnInit, OnDestroy {
     });
   }
 
+  onEdit() {
+    this.editForm.setValue({
+      title: this.task?.title as string,
+      description: this.task?.description as string,
+      type: this.task?.type as string,
+      status: this.task?.status as boolean
+    });
+
+    this.editable = !this.editable;
+  }
+
+  onCancel() {
+    this.editForm.reset();
+
+    this.editable = !this.editable;
+  }
+
+  onSave() {
+    if (this.editForm.valid) {
+      const editedTask: Task = {
+        id: this.task?.id as string,
+        title: this.editForm.value.title as string,
+        description: this.editForm.value.description as string,
+        type: this.editForm.value.type as string,
+        createdOn: this.task?.createdOn as Date,
+        status: this.editForm.value.status as boolean,
+      }
+
+      const id = this.task?.id;
+
+      if (id) {
+        this.taskService.updateTask(id, editedTask)
+        .pipe(takeUntil(this.destroy$))
+        .subscribe(() => {
+          this.ngOnInit();
```

Comment:
It is not a good practice to call lifecycle methods explicitly, because we expect to run it only once. If you want to call functionality from this method, it is better to create another method which will be called from both ngOnInit and  updateTask callback.
Btw, you have memory leak on line 39

Reply:
I read that you dont have to unsubscribe from activated route. from angular docs: The ActivatedRoute and its observables are insulated from the Router itself. The Router destroys a routed component when it is no longer needed and the injected ActivatedRoute dies with it.